com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceOperationsExecution.java

```diff
@@ -73,157 +78,72 @@ public AsyncExecutionState execute(ExecutionWrapper execution) {
     }
 
     protected List<CloudServiceExtended> getServicesWithTriggeredOperations(Collection<CloudServiceExtended> services,
-        Map<String, ServiceOperationType> triggeredServiceOperations) {
+                                                                            Map<String, ServiceOperationType> triggeredServiceOperations) {
         return services.stream()
-            .filter(e -> triggeredServiceOperations.containsKey(e.getName()))
-            .collect(Collectors.toList());
+                       .filter(e -> triggeredServiceOperations.containsKey(e.getName()))
+                       .collect(Collectors.toList());
     }
 
-    protected abstract List<CloudServiceExtended> computeServicesToPoll(ExecutionWrapper execution,
-        Map<String, ServiceOperationType> triggeredServiceOperations);
-
-    protected abstract ServiceOperation getLastServiceOperation(ExecutionWrapper execution, CloudControllerClient client,
-        CloudServiceExtended service);
+    protected List<CloudServiceExtended> computeServicesToPoll(ExecutionWrapper execution,
+                                                               Map<String, ServiceOperationType> triggeredServiceOperations) {
+        List<CloudServiceExtended> servicesData = getServicesData(execution.getContext());
+        return getServicesWithTriggeredOperations(servicesData, triggeredServiceOperations);
+    }
 
-    private ServiceOperation getLastServiceOperationAndHandleExceptions(ExecutionWrapper execution, CloudControllerClient client,
-        CloudServiceExtended service) {
+    private ServiceOperation getLastServiceOperationAndHandleExceptions(ExecutionWrapper execution, CloudServiceExtended service) {
```

Comment:
Containing the *And* in the method name indicates that this method is doing more than one thing.
Try to refactor it?